packages/angular-material/test/object-control.spec.ts

```diff
@@ -0,0 +1,209 @@
+    expect(ObjectControlRendererTester(uischema1, schema1)).toBe(2);
+    expect(ObjectControlRendererTester(uischema2, schema2)).toBe(2);
+  });
+});
+describe('Object Control', () => {
+  let fixture: ComponentFixture<any>;
+  let component: any;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        ObjectControlRenderer,
+        TextControlRenderer,
+        VerticalLayoutRenderer
+      ],
+      imports: [
+        CommonModule,
+        JsonFormsModule,
+        NgReduxModule,
+        MatCardModule,
+        NoopAnimationsModule,
+        MatFormFieldModule,
+        MatInputModule,
+        ReactiveFormsModule
+      ],
+      providers: [
+        { provide: NgRedux, useFactory: MockNgRedux.getInstance }
+      ]
+    }).overrideModule(BrowserDynamicTestingModule, {
+      set: {
+          entryComponents: [
+            TextControlRenderer,
+            VerticalLayoutRenderer
+          ]
+      }
+  })
+  .compileComponents();
+
+    MockNgRedux.reset();
+    fixture = TestBed.createComponent(ObjectControlRenderer);
+    component = fixture.componentInstance;
+  }));
+
+  it('render all elements', async(() => {
+    const mockSubStore = MockNgRedux.getSelectorStub();
+    component.uischema = uischema1;
+    component.schema = schema2;
+
+    mockSubStore.next({
+      jsonforms: {
+        renderers: renderers,
+        core: {
+          data: {},
+          schema: schema2,
+        }
+      }
+    });
+    component.ngOnInit();
+    mockSubStore.complete();
+    fixture.detectChanges();
+
+    fixture.whenRenderingDone().then(() => {
+      fixture.detectChanges();
+      expect(fixture.nativeElement.querySelectorAll('input').length).toBe(2);
+    });
+  }));
+
+  it('render only itself elements', async(() => {
+    const mockSubStore = MockNgRedux.getSelectorStub();
+    component.uischema = uischema2;
+    component.schema = schema2;
+
+    mockSubStore.next({
+      jsonforms: {
+        renderers: renderers,
+        core: {
+          data: {},
+          schema: schema2,
+        }
+      }
+    });
+    mockSubStore.complete();
+    fixture.detectChanges();
+    component.ngOnInit();
+    fixture.whenStable().then(() => {
+      expect(fixture.nativeElement.querySelectorAll('input').length).toBe(1);
+    });
+  }));
+
+  xit('can be disabled', async(() => {
+    const mockSubStore = MockNgRedux.getSelectorStub();
+    component.uischema = uischema1;
+    component.schema = schema1;
+    component.disabled = true;
+
+    mockSubStore.next({
+      jsonforms: {
+        renderers: renderers,
+        core: {
+          data: {},
+          schema: schema1,
+        }
+      }
+    });
+    mockSubStore.complete();
+    fixture.detectChanges();
+    component.ngOnInit();
+    fixture.whenStable().then(() => {
+      expect(fixture.nativeElement.querySelector('input').disabled).toBe(true);
+    });
+  }));
+  it('should be enabled by default', async(() => {
```

Comment:
Does this test actually test something, or put another way: can an object renderer propagate a disabled state to its children?

Reply:
I think it would be great if this would work, but I'm not sure whether it is doable and whether we want to support it.